### PR TITLE
Add support for Prometheus metrics as source of scaling

### PR DESCRIPTION
**Pull request by alexander-falca**
## Description
Add support for Prometheus metrics as source of scaling

## How Has This Been Tested?
Tested locally using `mage operator:test` 
Tested on EKS cluster, running operator locally `mage operator:run` and pointing to K8S API server 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [X] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [X] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] I have read the **CONTRIBUTING** document.
- [X] I have added tests to cover my changes.
- [X] All new and existing tests passed.

Note: Kubernetes pod autoscaler source code: https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/podautoscaler/horizontal.go

### CODE REVIEW

File: PROJECT [hunk 1]
```diff
@@ -0,0 +1,11 @@
+domain: core.adobe.com
+layout: go.kubebuilder.io/v2
+projectName: kratos
```
Comment (amuraru) [inline PROJECT:3]: `operator-sdk init` complains this `projectName` is unknown field. better to remove

File: config/crd/scaling.core.adobe.com_kratos.yaml [hunk 2]
```diff
@@ -1,10 +1,10 @@
 
 ---
-apiVersion: apiextensions.k8s.io/v1beta1
+apiVersion: apiextensions.k8s.io/v1
```
Comment (amuraru) [inline config/crd/scaling.core.adobe.com_kratos.yaml:3]: small nitpick: running `mage operator:generateCrds` with controller-gen 0.4.0 seems to generate some diffs in the generated yaml. better to double check

**Review by amuraru — APPROVED** [review]
nice!

File: config/crd/scaling.core.adobe.com_kratos.yaml (hunk 2 above)
Comment (alexander-falca) [reply config/crd/scaling.core.adobe.com_kratos.yaml:3]: thanks! checked, works as expected. I will keep eye on this.

File: PROJECT (hunk 1 above)
Comment (alexander-falca) [reply PROJECT:3]: I was unable to reproduce it locally, but I'm using latest operator-sdk@1.1.0 and controller-get@0.4.0

**Comment by alexander-falca** [comment]
Thanks for review @amuraru!

**Pull request merged**